src/misc/cross.cpp

```diff
@@ -439,6 +440,224 @@ FILE *fopen_wrap_ro_fallback(const std::string &filename, bool &is_readonly)
 	return fp;
 }
 
+bool wild_match(const char *haystack, char *needle)
+{
+	size_t max, i;
+	for (; *needle != '\0'; ++needle) {
+		switch (*needle) {
+		case '?':
+			if (*haystack == '\0')
+				return false;
+			++haystack;
+			break;
+		case '*':
+			if (needle[1] == '\0')
+				return true;
+			max = strlen(haystack);
+			for (i = 0; i < max; i++)
```

Comment:
If I is only assigned here, let's create it here in the loop.

Reply:
Done.